acceptance/analyzer_test.go

```diff
@@ -76,317 +77,500 @@ func TestAnalyzer(t *testing.T) {
+						"--env", "CNB_PLATFORM_API="+apiString,
+					),
+					h.WithArgs(
+						ctrPath(analyzerPath),
+						"-cache-dir", ctrPath("/cache"),
+						"-group", ctrPath("/layers/other-group.toml"),
+						"some-image",
+					),
+				)
 
-			it.Before(func() {
-				appImage = "some-app-image-" + h.RandString(10)
-				metadata := minifyMetadata(t, filepath.Join("testdata", "analyzer", "app_image_metadata.json"), platform.LayersMetadata{})
+				h.AssertPathExists(t, filepath.Join(copyDir, "layers", "some-other-buildpack-id"))
+				h.AssertPathDoesNotExist(t, filepath.Join(copyDir, "layers", "some-buildpack-id"))
+			})
+		})
 
-				cmd := exec.Command(
-					"docker",
-					"build",
-					"-t", appImage,
-					"--build-arg", "fromImage="+containerBaseImage,
-					"--build-arg", "metadata="+metadata,
-					filepath.Join("testdata", "analyzer", "app-image"),
+		when("analyzed path is provided", func() {
+			it("writes analyzed.toml at the provided path", func() {
+				h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) < 0, "Platform API < 0.7 accepts an image argument")
+				h.DockerRunAndCopy(t,
+					containerName,
+					copyDir,
+					ctrPath("/some-dir/some-analyzed.toml"),
+					analyzeImage,
+					h.WithFlags("--env", "CNB_PLATFORM_API="+apiString),
+					h.WithArgs(
+						ctrPath(analyzerPath),
+						"-analyzed", ctrPath("/some-dir/some-analyzed.toml"),
+						"-previous-image", "some-image",
+					),
 				)
-				h.Run(t, cmd)
+
+				assertAnalyzedMetadata(t, filepath.Join(copyDir, "some-analyzed.toml"))
 			})
 
-			it.After(func() {
-				h.DockerImageRemove(t, appImage)
+			it("writes analyzed.toml at the provided path", func() {
+				h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not accept an image argument")
+				h.DockerRunAndCopy(t,
+					containerName,
+					copyDir,
+					ctrPath("/some-dir/some-analyzed.toml"),
+					analyzeImage,
+					h.WithFlags("--env", "CNB_PLATFORM_API="+apiString),
+					h.WithArgs(
+						ctrPath(analyzerPath),
+						"-analyzed", ctrPath("/some-dir/some-analyzed.toml"),
+						"some-image",
+					),
+				)
+
+				assertAnalyzedMetadata(t, filepath.Join(copyDir, "some-analyzed.toml"))
+			})
+		})
+
+		when("daemon case", func() {
+			it("writes analyzed.toml", func() {
+				h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7 does not accept an image argument")
+				h.DockerRunAndCopy(t,
+					containerName,
+					copyDir,
+					ctrPath("/layers/analyzed.toml"),
+					analyzeImage,
+					h.WithFlags(append(
+						dockerSocketMount,
+						"--env", "CNB_PLATFORM_API="+apiString,
+					)...),
+					h.WithArgs(
+						ctrPath(analyzerPath),
+						"-daemon",
+						"some-image",
+					),
+				)
+
+				assertAnalyzedMetadata(t, filepath.Join(copyDir, "analyzed.toml"))
 			})
 
-			it("restores app metadata", func() {
-				output := h.DockerRunAndCopy(t,
+			it("writes analyzed.toml", func() {
```

Comment:
The same comment as https://github.com/buildpacks/lifecycle/pull/582/files#r623979445 for these two `it`.

Reply:
6d7c623